pkg/controller/appdefinition/deploy.go

```diff
@@ -196,17 +198,26 @@ func toContainers(app *v1.AppInstance, tag name.Reference, name string, containe
 		})
 	}
 
-	containers = append(containers, toContainer(app, tag, name, name, container))
+	newContainer, err := toContainer(req, app, tag, name, name, container)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	containers = append(containers, newContainer)
 	for _, entry := range typed.Sorted(container.Sidecars) {
-		newContainer := toContainer(app, tag, name, entry.Key, entry.Value)
+		newContainer, err := toContainer(req, app, tag, name, entry.Key, entry.Value)
```

Comment:
nit: Don't reallocate on every loop iteration if you already have the necessary.
```suggestion
		newContainer, err = toContainer(req, app, tag, name, entry.Key, entry.Value)
```